proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,320 @@
+        im_blur = blur_op(x)
+    else:
+        im_blur = px.sum([
+            px.conv(psf_modes[i], px.mul_elemwise(weights[i], x))
+            for i in range(n_psf)
+        ])
+
+    # The weights are manually "absorbed" into the L2-norm, which has a fast
+    # elemwise proximal operator.
+    #
+    # TODO: Crop the im_stack and weight to double of image grid size; elminiate
+    # redundant 2D convolution with zero weights.
+    data_term = px.sum_squares(im_blur - im)
+
+    im_grad = px.grad(x)
+    grad_term = mu * alpha * n_psf * px.group_norm1(im_grad, group_dims=[
+        2
+    ]) + mu * (1 - alpha) * n_psf * px.sum_squares(im_grad)
+    return data_term + grad_term
+
+
+def draw(im, idx: int, title: str):
+    plt.subplot(3, 4, idx)
+    plt.imshow(im, cmap='gray')
+    plt.axis('off')
+    plt.title(title)
+
+
+def get_psf(n_psf: int, kwidth: int):
+    assert n_psf <= 4
+
+    psf_modes = np.empty((n_psf, kwidth, kwidth), dtype=np.float32)
+    psf_modes[0] = get_kernel(kwidth, 2)
+
+    for i in range(1, n_psf):
+        psf_modes[i] = np.rot90(psf_modes[0], k=i)
+
+    return psf_modes
+
+
+def get_mask(n_psf: int, imwidth: int, feather_edge: int = 100):
+    assert n_psf <= 4
+
+    mask = np.zeros((n_psf, imwidth, imwidth), dtype=np.float32)
+
+    W = int(imwidth / 2)
+    mask[0, :W, :W] = 1
+    mask[0] = gaussian_filter(mask[0], feather_edge)
+
+    for i in range(1, n_psf):
+        mask[i] = np.rot90(mask[0], k=i)
+
+    # Normalize the weig
+    mask[:] /= np.sum(mask, axis=0)[np.newaxis, ...]
+
+    return mask
+
+
+def simulate(im, psf_modes, mask, noise_level: float = 1e-2):
+    _x = px.Variable(im.shape)
+    blur_op = weighted_planewise_conv(psf_modes, mask, im.shape, n_psf)(_x)
+    im_sb = np.empty(im.shape, dtype=np.float32, order='F')
+    blur_op.forward([im], [im_sb])
+
+    im_sb += noise_level * im_sb.max() * np.random.randn(*im.shape)
+
+    return im_sb
+
+
+# Sample Data (synthetic or load from file)
+imwidth = 512
+kwidth = 11
+n_psf = 4
+
+raw = get_test_image(imwidth)
+draw(raw, 1, 'Raw')
+
+psf_modes = get_psf(n_psf, kwidth)
+mask = get_mask(n_psf, imwidth)
+measured = simulate(raw, psf_modes, mask)
+
+# Delete the raw image to ensure the information doesn't leak to the solver
+del raw
+
+# Define the problem
+mu, alpha, maxiter = 2e-3, 0.9, 200
+x = px.Variable(measured.shape)
+prob = px.Problem(
+    FlickerAndRigaut2005(x,
+                         measured,
+                         psf_modes,
+                         mask,
+                         mu=mu,
+                         alpha=alpha,
+                         n_psf=n_psf))
+
+prob.solve(solver='pc',
+           eps_abs=1e-3,
+           max_iters=maxiter,
+           conv_check=20,
+           verbose=True,
+           x0=measured)
+
+for i in range(4):
+    draw(psf_modes[i], 5 + i, f'camera shake[{i}]')
+    draw(mask[i], 9 + i, f'weight[{i}]')
+
+draw(measured, 2, 'measured')
+draw(x.value, 3, 'reconstructed')
+plt.show()
```

Comment:
Expected output:
![image](https://github.com/comp-imaging/ProxImaL/assets/12185630/e4895cb4-b29f-43da-807e-a2a959b159e0)

Reply:
For a more realistic image reconstruction results, scale the problem by `imwidth=2048`, and `kwidth=31`.

![image](https://github.com/comp-imaging/ProxImaL/assets/12185630/dd0aec81-dcb7-47d5-8e19-5e935d906038)